heron/common/src/python/network/heron_client.py

```diff
@@ -0,0 +1,314 @@
+# Copyright 2016 Twitter. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+'''heron_client.py'''
+
+import asyncore
+import socket
+import traceback
+from abc import abstractmethod
+
+import time
+from heron.common.src.python.network import HeronProtocol, REQID, StatusCode, OutgoingPacket
+from heron.common.src.python.log import Log
+import heron.common.src.python.constants as constants
+
+# pylint: disable=too-many-instance-attributes
+# pylint: disable=fixme
+class HeronClient(asyncore.dispatcher):
```

Comment:
The Heron Client is the term used to describe the heron command line client that end users use (i.e. the heron cli). This seems to be a multi-purpose networking client. Can we name this differently to avoid confusion?